odxtools/parameters/tablekeyparameter.py

```diff
@@ -71,7 +72,11 @@ def _resolve_snrefs(self, diag_layer: "DiagLayer") -> None:
 
     @property
     def table(self) -> "Table":
-        return self._table
+        if self._table is not None:
+            return self._table
+        if self._table_row is not None:
+            return self._table_row.table
+        odxraise(f'Could not resolve the table of {self.short_name}')
```

Comment:
I think it would be better to specify the return type of this property as `Optional["Table"]` and keep it always returning `self._table`...

Reply:
Tried that, got 13 mypy errors,

`self.table` is never `None`, but it may be resolved from `_table_row` instead of being referenced directly.

Reply:
In this case, I think that `self._table` should be declared to be `self.table: Optional["Table"]` and the static-case resolution should go to `._resolve_snrefs()`...

Reply:
How would the declaration in `__post_init__` look like? I can't figure out a mypy happy declaration for it

Reply:
the same as it does now, but without the assignment:
```python
    def __post_init__(self) -> None:
        self._table: Optional["Table"]
        # [...]
```